Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1507,18 +1507,10 @@ public string GetLocalApiUrl(IPAddress ipAddress)
         public string GetLocalApiUrl(ReadOnlySpan<char> host)
         {
             var url = new StringBuilder(64);
-            if (EnableHttps)
-            {
-                url.Append("https://");
-            }
-            else
-            {
-                url.Append("http://");
-            }
-
-            url.Append(host)
+            url.Append(EnableHttps ? "https://" : "http://")
+                .Append(host)
                 .Append(':')
-                .Append(HttpPort);
+                .Append(EnableHttps ? HttpsPort : HttpPort);
```

Comment:
You should also check `Certificate != null`, you can see the condition checked to enable HTTPS here: https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Server/Program.cs#L277

Reply:
while I agree, this `appHost.EnableHttps` is used throughout the codebase without checking for certificate... what's the best way of tackling this?

Reply:
Create a new property `IsHttpsEnabled` that returns `EnableHttps && Certificate != null`?
And possibly remove `EnableHttps` from apphost and only have it in the config

Reply:
I wanted to make that property, but I fear a lot of stuff would break... but let's try :)

Reply:
Hmmm....
```cs
        public bool EnableHttps => SupportsHttps && ServerConfigurationManager.Configuration.EnableHttps;

        public bool SupportsHttps => Certificate != null || ServerConfigurationManager.Configuration.IsBehindProxy;
```
it's already there (not there only if "handled by reverse proxy" is enabled, but assuming https in local api url is in line with what other things are doing anyway).

Reply:
Ah, good catch. Those properties are confusing -_-